src/libmongoc/src/mongoc/mongoc-cluster-private.h

```diff
@@ -43,14 +43,14 @@ BSON_BEGIN_DECLS
 typedef struct _mongoc_cluster_node_t {
    mongoc_stream_t *stream;
    char *connection_address;
+   uint32_t generation;
 
+   /* TODO CDRIVER-3653, these fields are unused. */
```

Comment:
This is a general issue in the driver. A `mongoc_server_stream_t` is a `mongoc_stream_t` with a server description  attached. We attach that server description when a stream is fetched. But in between server selection and fetching the stream, another thread could mark the server description as Unknown. That has the unfortunate consequence of making wire version checks think we have maxWireVersion 0.

I think we should fix this, but this seemed out of scope for this change.